IdentityCore/src/util/ASAuthorizationSingleSignOnProvider+MSIDExtensions.m

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+#import "ASAuthorizationSingleSignOnProvider+MSIDExtensions.h"
+#import "MSIDConstants.h"
+
+@implementation ASAuthorizationSingleSignOnProvider (MSIDExtensions)
+
++ (ASAuthorizationSingleSignOnProvider *)msidSharedProvider
+{
+    static dispatch_once_t once;
+    static ASAuthorizationSingleSignOnProvider *ssoProvider;
+    
+    dispatch_once(&once, ^{
+        NSURL *url = [NSURL URLWithString:MSID_DEFAULT_AAD_AUTHORITY];
+    
+        ssoProvider = [ASAuthorizationSingleSignOnProvider authorizationProviderWithIdentityProviderURL:url];
```

Comment:
What if SSO extension presence status changes during process lifetime? Is ASAuthorizationSingleSignOnProvider smart enough in following scenario:

1. ASAuthorizationSingleSignOnProvider is created before SSO extension is activated, so canPerformAuthorization == NO
2. SSO extension got activated
3. Will same instance of ASAuthorizationSingleSignOnProvider now return canPerformAuthorization == YES?

Also, what are other pros and cons of using single shared instance vs creating new instance every time? I imagine, creating new instance should be relatively cheap. But if creation fails, we have a permanent failure for the lifetime of the process vs just for one request.

Reply:
removed singleton